src/esp/assets/ResourceManager.cpp

```diff
@@ -148,8 +150,8 @@ void ResourceManager::initPhysicsManager(
   if (isEnabled) {
     if (physicsManagerAttributes->getSimulator().compare("bullet") == 0) {
 #ifdef ESP_BUILD_WITH_BULLET
-      physicsManager.reset(
-          new physics::BulletPhysicsManager(*this, physicsManagerAttributes));
+      physicsManager = std::make_shared<physics::BulletPhysicsManager>(
+          *this, physicsManagerAttributes);
```

Comment:
Pretty sure we need this and the assignment below to be a reset, to retain the same ptr.  @aclegg3?

Reply:
Yeah, won't this re-assign the particular shared_ptr instance to a new PhysicsManager while leaving any other references alive? We don't want that. We want to change the data all shared_ptrs are referencing here.

Reply:
If you want to change the data all shared_ptrs are referencing you need to do `*physicsManager = ...`.  Reset only impacts this specific shared_ptr instance, as does swap.

Reply:
```cpp
int main() {
    auto a = std::make_shared<int>(5);
    auto b = a;

    a.reset(new int{3});

    std::cout << *a << std::endl << *b << std::endl;
}
```
```
3
5
```

Reply:
https://en.cppreference.com/w/cpp/memory/shared_ptr/operator%3D

Reply:
Yeah @erikwijmans I was pretty sure it was equivalent behavior. @aclegg3 Given this info, what is the diff suppose to be?

Reply:
doing `*physics_manager=` works fine for BulletPhysicsManager, but when I construct a normal physics manager and try to set it to that I get errors.

Reply:
The wrapper object managers maintain a weak ptr built from the original shared physicsManager ptr.  Will these weak pointers still track the physicsManager::ptr after its re-assignment?  This was why I was thinking the reset might still be needed.

Reply:
Sorry, my mistake. This was not originally meant to reset all smart pointer references. Only Simulator has an active reference to the object. We use shared_ptr because the object managers will have a weak_ptr to the PhysicsManager to do their work. It should be fine to re-assign this referenced smart pointer as Aaron originally suggested.